neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,68 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            filename=None,
+            exclude_filename=None,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if not dirname:
+            raise ValueError("`dirname` cannot be empty.")
+
+        if filename:
+            include_filenames = filename
+            warnings.warn("`filename` is deprecated and will be removed. Please use `include_filenames` instead")
+
+        if exclude_filename:
+            exclude_filenames = exclude_filename
+            warnings.warn("`exclude_filename` is deprecated and will be removed. Please use `exclude_filenames` instead")
+
+        if include_filenames is None:
+            include_filenames = []
+        elif not isinstance(include_filenames, (list, set, np.ndarray)):
+            include_filenames = [include_filenames]
+
+        if exclude_filenames is None:
+            exclude_filenames = set()
+        elif not isinstance(exclude_filenames, (list, set, np.ndarray)):
+            exclude_filenames = {exclude_filenames}
+
+        if include_filenames:
+            include_filepath = {os.path.dirname(f) for f in include_filenames}
+            if len(include_filepath) > 1:
+                raise ValueError("Files in include_filename must be in a single path!")
```

Comment:
This check isn't necessary because we force the user to give us the dirname they want us to look in. We should do that check later.
```suggestion
```